src/checkbox/tests/unit/Checkbox.ts

```diff
@@ -1,151 +1,230 @@
+					'aria-readonly': String(checkboxProperties.readOnly),
+					bind: widget,
+					checked: checkboxProperties.checked,
+					classes: widget.classes(css.input),
+					disabled: checkboxProperties.disabled,
+					name: checkboxProperties.name,
+					onblur: widget.listener,
+					onchange: widget.listener,
+					onclick: widget.listener,
+					onfocus: widget.listener,
+					onmousedown: widget.listener,
+					onmouseup: widget.listener,
+					ontouchstart: widget.listener,
+					ontouchend: widget.listener,
+					ontouchcancel: widget.listener,
+					readOnly: checkboxProperties.readOnly,
+					required: checkboxProperties.required,
+					type: 'checkbox',
+					value: checkboxProperties.value
+				})
+			])
+		];
+
+		widget.setProperties(checkboxProperties);
 
-	'state classes'() {
+		// TODO: is there a better way to do this than a throwaway Checkbox instance?
 		const checkbox = new Checkbox();
-		checkbox.setProperties({
-			checked: true,
-			disabled: true,
-			invalid: true,
-			readOnly: true,
-			required: true,
-			mode: Mode.toggle,
-			offLabel: 'Off',
-			onLabel: 'On'
-		});
-		let vnode = <VNode> checkbox.__render__();
-
-		assert.isTrue(vnode.properties!.classes![css.checked]);
-		assert.isTrue(vnode.properties!.classes![css.disabled]);
-		assert.isTrue(vnode.properties!.classes![css.invalid]);
-		assert.isTrue(vnode.properties!.classes![css.readonly]);
-		assert.isTrue(vnode.properties!.classes![css.required]);
-		assert.isTrue(vnode.properties!.classes![css.toggle]);
-
-		checkbox.setProperties({
-			checked: false,
-			disabled: false,
+		widget.expectRender(w<LabelProperties>(Label, {
+			classes: checkbox.classes(css.root, css.toggle, css.checked, css.disabled, css.invalid, css.readonly, css.required),
+			formId: checkboxProperties.formId,
+			label: checkboxProperties.label!
+		}, children));
+	},
+
+	// TODO: is it preferred to combine this with the previous test?
+	// Manipulating the harness and the expected results is more hassle than it's worth
+	'properties and attributes 2'() {
+		const children: HNode[] = [
+			v('div', {
+				classes: widget.classes(css.inputWrapper)
+			}, [
+				v('div', { classes: widget.classes(css.onLabel) }, [ null ]),
+				v('div', { classes: widget.classes(css.offLabel) }, [ null ]),
+				v('input', {
+					'aria-describedby': undefined,
+					'aria-invalid': null,
+					'aria-readonly': null,
+					bind: widget,
+					checked: false,
+					classes: widget.classes(css.input),
+					disabled: undefined,
+					name: undefined,
+					onblur: widget.listener,
+					onchange: widget.listener,
+					onclick: widget.listener,
+					onfocus: widget.listener,
+					onmousedown: widget.listener,
+					onmouseup: widget.listener,
+					ontouchstart: widget.listener,
+					ontouchend: widget.listener,
+					ontouchcancel: widget.listener,
+					readOnly: undefined,
+					required: undefined,
+					type: 'checkbox',
+					value: undefined
+				})
+			])
+		];
```

Comment:
Could you define `children` once above, and use `assignProperties()` or `assignChildProperties()` when necessary to change specific properties per test (probably necessary at least for classes)? Should make things more concise, at least.